src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/Export/CosmosExportOrchestratorJob.cs

```diff
@@ -36,7 +44,43 @@ public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progre
 
             var groupJobs = await _queueClient.GetJobByGroupIdAsync(QueueType.Export, jobInfo.GroupId, true, cancellationToken);
 
-            if (groupJobs.Count == 1)
+            // Parallel system level export is parallelized by resource type and CosmosDB physical partitions feed range.
+            if (record.ExportType == ExportJobType.All && record.IsParallel && (record.Filters == null || record.Filters.Count == 0))
+            {
+                var resourceTypes = string.IsNullOrEmpty(record.ResourceType)
+                                 ? (await _searchService.GetUsedResourceTypes(cancellationToken))
+                                 : record.ResourceType.Split(',');
+
+                var physicalPartitionFeedRanges = await _searchService.GetFeedRanges(cancellationToken);
+
+                var enqueuedRangesByResourceType = groupJobs.Select(x => JsonConvert.DeserializeObject<ExportJobRecord>(x.Definition))
+                                                    .Where(x => x.ResourceType is not null)
+                                                    .GroupBy(x => x.ResourceType)
+                                                    .ToDictionary(x => x.Key, x => x.Select(x => x.FeedRange).ToList());
+
+                foreach (var resourceType in resourceTypes)
+                {
+                    // Skip any feed range/resource type combos that have already been queued.
+                    // This scenario is when the coordinator is killed during queueing of export jobs.
+                    var notEnqueued = enqueuedRangesByResourceType.TryGetValue(resourceType, out var enqueuedRangesThisResourceType) ?
+                                            physicalPartitionFeedRanges.Where(x => !enqueuedRangesThisResourceType.Contains(x)) :
+                                            physicalPartitionFeedRanges;
+
+                    foreach (var partitionRange in notEnqueued)
+                    {
+                        var processingRecord = CreateExportRecord(
+                                record,
+                                jobInfo.GroupId,
+                                resourceType: resourceType,
+                                feedRange: partitionRange);
+
+                        string[] definitions = [JsonConvert.SerializeObject(processingRecord)];
+
+                        await _queueClient.EnqueueAsync((byte)QueueType.Export, definitions, jobInfo.GroupId, false, false, cancellationToken);
```

Comment:
To make sure I'm understanding this correctly: one job is being created for every resource type / physical partition combination?

Reply:
Correct - these will queue up have different execution periods. If a database is heavy on Observations, the observation export will take the longest to run (depending on partition distribution). The larger the database the more parallel the job can execute.